ResearchSuite/ResearchSuite/RSDConditionalStepNavigator.swift

```diff
@@ -242,11 +242,30 @@ extension RSDConditionalStepNavigator {
     
     public func progress(for step: RSDStep, with result: RSDTaskResult?) -> (current: Int, total: Int, isEstimated: Bool)? {
         if let markers = self.progressMarkers {
-            guard let stepHistory = result?.stepHistory.map({ $0.identifier }), let current = markers.lastIndex(where: { stepHistory.contains($0) }) else {
+            // Get the list of steps that have been shown and add the step under test in case this is
+            // called before that step is added to the step history.
+            guard let stepHistory = result?.stepHistory.map({ $0.identifier }) else { return nil }
+            var stepList = stepHistory
+            stepList.append(step.identifier)
```

Comment:
Should you check if it's already there first? Or does it not matter if it ends up on there twice?